### PR TITLE
CORE-316 Update quicklaunches endpoints to use app versions

### PR DESCRIPTION
**Pull request by psarando**
This PR will add `app_version_id` to `/quicklaunches` CRUD endpoint requests and responses.

PRs cyverse-de/de-database#10 and cyverse-de/common-swagger-api#75 must be merged first.

### CODE REVIEW

File: src/analyses/controllers.clj [hunk 1]
```diff
@@ -6,16 +6,19 @@
...
 (defn add-quicklaunch
   [user quicklaunch]
-  (clients/validate-submission
-   {:quicklaunch quicklaunch
-    :app         (clients/get-app user system-id (:app_id quicklaunch))
-    :system-id   system-id
-    :user        user})
-  (persist/add-quicklaunch user quicklaunch))
+  (let [{:keys [version_id] :as app} (clients/get-app user system-id (:app_id quicklaunch))
+        quicklaunch                  (assoc quicklaunch :app_version_id version_id)]
```
Comment (slr71) [inline src/analyses/controllers.clj:10]: This is more of a question than a comment. Is the purpose of this PR simply to get the service working with the updated schema?

My concern was that if the app version isn't included in the quick launch definition request then the parameter IDs will probably change between app versions. I suspect that this concern won't apply in the final version of the code, however.

**Review by slr71 — COMMENTED** [review]
I ran out of time for the day; I should be able to finish the review tomorrow, though.

File: src/analyses/controllers.clj (hunk 1 above)
Comment (psarando) [reply src/analyses/controllers.clj:10]: That's correct on both counts.

This round of PRs is only to get the services compatible with the schema changes in cyverse-de/de-database#10, without new requirements in endpoint requests or in the UI. This usually means using the latest version of an app in the services for now.

I agree that eventually this endpoint will need to require the app version ID in the request.

Comment (slr71) [reply src/analyses/controllers.clj:10]: I thought that was probably the case. Thanks for the confirmation.

File: src/analyses/controllers.clj [hunk 2]
```diff
@@ -6,16 +6,19 @@
...
+      {:quicklaunch quicklaunch
+       :app         app
+       :system-id   system-id
+       :user        user})
+    (persist/add-quicklaunch user quicklaunch)))
 
 (defn update-quicklaunch
   [id user quicklaunch]
   (let [ql-merged  (merge (persist/get-quicklaunch id user) quicklaunch)
+        ;FIXME     Lookup by app version ID
```
Comment (slr71) [inline src/analyses/controllers.clj:21]: If I'd gotten this far in the review yesterday, I would have realized that more stuff was coming later. 😆 🤦‍♀️

**Review by slr71 — APPROVED** [review]
Looks good to me! :shipit:

**Review by johnworth — APPROVED** [review]
:shipit: looks good to me.

**Comment by psarando** [comment]
Thanks for the reviews! 👍

**Pull request merged**